File: qb-engineer-server/qb-engineer.core/Models/CreateAssetRequestModel.cs
using QBEngineer.Core.Enums;

namespace QBEngineer.Core.Models;

public record CreateAssetRequestModel(
    string Name,
    AssetType AssetType,
    string? Location,
    string? Manufacturer,
    string? Model,
    string? SerialNumber,
    string? Notes);

File: qb-engineer-ui/src/app/features/assets/models/create-asset-request.model.ts
import { AssetType } from './asset-type.type';

export interface CreateAssetRequest {
  name: string;
  assetType: AssetType;
  location?: string;
  manufacturer?: string;
  model?: string;
  serialNumber?: string;
  notes?: string;
  isCustomerOwned?: boolean;
  cavityCount?: number;
  toolLifeExpectancy?: number;
}
